Include/Rover/Model.hpp

```diff
@@ -0,0 +1,190 @@
+#ifndef ROVER_MODEL_HPP
+#define ROVER_MODEL_HPP
+#include <algorithm>
+#include <numeric>
+#include <random>
+#include <type_traits>
+#include <vector>
+#include "ScalarView.hpp"
+
+namespace {
+  template<typename R, typename X, typename Y>
+  constexpr std::enable_if_t<!std::is_same_v<std::decay_t<R>,
+      std::decay_t<X>> || !std::is_same_v<std::decay_t<R>, std::decay_t<Y>>,
+      bool> solve_basis(R&, X&, Y&) {
+    return false;
+  }
+
+  template<typename R, typename X, typename Y>
+  std::enable_if_t<std::is_same_v<std::decay_t<R>, std::decay_t<X>> &&
+      std::is_same_v<std::decay_t<R>, std::decay_t<Y>>, bool>
+      solve_basis(R& r, X& x, Y& y) {
+    if(x == y) {
+      return false;
+    }
+    if(x + y == x) {
+      r = x;
+    }
+    else {
```

Comment:
Use the one-true-brace style.

Reply:
Doesn't the one-true-brace style refer to keeping curly braces even if there is a single statement?